test/general/header_inclusion_order_async_0.pass.cpp

```diff
@@ -34,7 +34,7 @@ main()
     std::vector<T, allocator> data(n, 1, alloc);
 
     auto f = oneapi::dpl::experimental::reduce_async(TestUtils::make_device_policy(q), data.begin(), data.end());
-    f.wait();
+    f.deferrable_wait();
```

Comment:
As mentioned above, I'm not sure we want users calling this function, but rather they would be calling `wait()`

Reply:
@akukanov how do you think, which call we should use in this case?

Reply:
Fixed: restored `wait()` call.

Reply:
I recommend at least uglifying it, but we could consider having an `__internal_future` type with this functionality, and another which we return to users which can be created from an `__internal_future` which does not have this functionality.

Reply:
Uglified.